### ISSUE TITLE
Add LispNumberOrMarker

### ISSUE THREAD

**Issue by brotzeit** [Wilfred/remacs#1110]
`lisp_fn` also allows markers for `LispNumber` arguments. We need an additional type `LispNumberOrMarker` to finish the work of #983.

**Comment by carlosmn** [comment]
Is the idea that instead of doing the coercion in `From<LispObject> for Option<LispNumber>` we would instead accept `LispNumberOrMarker` so we're able to differentiate the origin? Otherwise the goal of accepting either a number or a marker seems to have been achieved in the referenced PR.

**Comment by brotzeit** [comment]
@carlosmn As you said, we only need this to indicate a LispNumber can also be a marker.

**Comment by krobelus** [comment]
Do we want to have `LispNumber` (not accepting markers) and `LispNumberOrMarker`?

**Comment by brotzeit** [comment]
@krobelus yeah. I think`LispNumberOrMarker` will be only relevant for lisp functions.

**Comment by brotzeit** [comment]
We should also verify that we used `LispNumber` correctly as `lisp_fn` arguments. I'm not so sure about that.
